biodatasets/osiris/osiris.py

```diff
@@ -0,0 +1,307 @@
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir[0]),
+                    "split": "data",
+                },
+            )
+        ]
+
+    def _get_offsets(self, parent: ET.Element, child: ET.Element) -> List[int32]:
+        '''
+        Retrieves character offsets for child from parent.
+        '''
+        parent_text = " ".join([" ".join([t for t in c.itertext()])
+                               for c in list(parent) if c.tag != "Pmid"])
+        child_text = " ".join([t for t in child.itertext()])
+        start = parent_text.index(child_text)
+        end = start + len(child_text)
+        return [start, end]
+
+    def _get_dict(self, elem: ET.Element) -> dict:
+        '''
+        Retrieves dict from XML element.
+        '''
+        elem_d = dict()
+        for child in elem:
+            elem_d[child.tag] = {}
+            elem_d[child.tag]["text"] = " ".join([t for t in child.itertext()])
+
+            if child.tag != "Pmid":
+                elem_d[child.tag]["offsets"] = self._get_offsets(elem, child)
+
+            for c in child:
+                elem_d[c.tag] = []
+
+            for c in child:
+                c_dict = c.attrib
+                c_dict["offsets"] = self._get_offsets(elem, c)
+                elem_d[c.tag].append(c.attrib)
+
+        return elem_d
+
+    def _handle_missing_variants(self, row: dict) -> dict:
+        '''
+        If variant is not present in the row this function adds one variant 
+        with no data (to make looping though items possible) and returns the new row.
+        These mocked variants will be romoved after parsing.
+        Otherwise returns unchanged row.
+        '''
+
+        if row.get("variant", 0) == 0:
+            row["variant"] = [
+                {
+                    "v_id": "",
+                    "v_lex": "",
+                    "v_norm": "",
+                    "offsets": [0, 0]
+                }
+            ]
+        return row
+
+    def _get_entities(self, row: dict) -> List[dict]:
+        '''
+        Retrieves two lists of dicts for genes and variants. 
+        After that, chains both together.
+        '''
+        genes = [
+            {
+                "id": str(uuid.uuid4()),
+                "offsets": [gene["offsets"]],
+                "text": [gene["g_lex"]],
+                "type": "gene",
+                "normalized":  [{"db_name": "NCBI Gene", "db_id": gene["g_id"]}]
+            } for gene in row["gene"]
+        ]
+
+        variants = [
+            {
+                "id": str(uuid.uuid4()),
+                "offsets": [variant["offsets"]],
+                "text": [variant["v_lex"]],
+                "type": "variant",
+                "normalized":  [{"db_name": "dbSNP" if variant["v_id"] == "No" else "NCBI Gene", "db_id": variant["v_norm"] if variant["v_id"] == "No" else variant["v_id"]}]
```

Comment:
Could you please change this to
```python
"db_name": "HGVS-like" if variant["v_id"] == "No" else "dbSNP"
```
Variants are normalized either to `dbSNP` if `v_id` is a number, otherwise to the HGVS notation if `v_id="No`

Reply:
Done. Thank you!